sonoff.py

```diff
@@ -51,35 +46,57 @@ def devices(self):
             'version': self.version,
             'nonce': str(self.ts) 
             }
-        r = requests.get(self.baseurl + 'api/user/device',params=payload,headers={'Authorization': self.auth})
+        r = requests.get(self.baseurl + 'api/user/device', params=payload, headers={'Authorization': self.auth})
         devices = []
         for i in r.json()['devicelist']:
-            d = {'name': i['name'],
-                'deviceid': i['deviceid'],
-                'status': i['params']['switch'],
-                # 'onlineTime': i['onlineTime'], TODO format as YYYY-MM-DD HH:MM:SS with tz
-                # 'offlineTime': i['offlineTime'
-                }
-            devices.append(d)
-        self.devices = devices
-        print(devices)
+            if 'switches' in i['params']:
+                for switch in i['params']['switches']:
+                    d = {'name': i['name'],
+                         'deviceid': i['deviceid'],
+                         'outletid': switch['outlet'],
+                         'status': switch['switch'],
+                    # 'onlineTime': i['onlineTime'], TODO format as YYYY-MM-DD HH:MM:SS with tz
+                    # 'offlineTime': i['offlineTime'
+                         }
+                    devices.append(d)
 
+            else: #Single switch devices
+                d = {'name': i['name'],
+                     'deviceid': i['deviceid'],
+                     'outletid': 0,
+                     'status': i['params']['switch'],
+                    # 'onlineTime': i['onlineTime'], TODO format as YYYY-MM-DD HH:MM:SS with tz
+                    # 'offlineTime': i['offlineTime'
+                    }
+                devices.append(d)
+        self.devices = devices
 
-    def change_device_status(self, deviceid:str, new_status:str):
-        self.status = [stat.get('status') for stat in self.devices if stat.get('deviceid') == deviceid][0]
+    def change_device_status(self, deviceid: str, outletid: int, new_status: str):
```

Comment:
Can you add pls default value to outletid? --> change_device_status(self, deviceid: str, new_status: str, outletid: int=0)

Reply:
Good as done